lib/output.js

```diff
@@ -117,15 +117,37 @@ function OutputStream(options) {
         function quote_double() {
             return '"' + str.replace(/\x22/g, '\\"') + '"';
         }
+        function quote_template() {
+            if (!options.ascii_only) {
+                str = str.replace(/\\(r|n|u2028|u2029)/g, function(s, c) {
+                    switch(c) {
+                        case "n": return "\n";
+                        case "r": return "\r";
+                        case "u2028": return "\u2028";
+                        case "u2029": return "\u2029";
+                    }
+                    return "\\" + c;
+                });
+            }
+            return '`' + str.replace(/`/g, '\\`') + '`';
+        }
         if (options.ascii_only) str = to_ascii(str);
         switch (options.quote_style) {
           case 1:
+            if (quote == "`" && sq > 0)
+                return quote_template();
             return quote_single();
           case 2:
+            if (quote == "`" && dq > 0)
+                return quote_template();
             return quote_double();
           case 3:
+            if (quote == "`")
+                return quote_template();
             return quote == "'" ? quote_single() : quote_double();
           default:
+            if (quote == "`" && dq > 0 && sq > 0)
```

Comment:
This example makes the problem more clear:
```
$ cat test.js
f(`Hello`);
f('Hello');
f("Hello");

g(`Hello'`);
g('Hello\'');
g("Hello'");
```
```
$ bin/uglifyjs -c sequences=0 -b quote_style=1 -- test.js
f('Hello');    // <--- notice this is single quoted
f('Hello');
f('Hello');

g(`Hello'`);   // <--- should also be single quoted
g('Hello\'');
g('Hello\'');
```
Whether `sq > 0` or `dq > 0` is irrelevant when `quote_style` is `1` or `2`.

As for `quote_style=0`, the line:
```
            if (quote == "`" && dq > 0 && sq > 0)
```
should be:
```
            if (quote == "`" && bq < dq && bq < sq)
```
where `bq` is the number of back quotes in the string.